src/main/java/ca/softwarespace/qiyanna/dataaggregator/repositories/sql/SQLLeagueEntryRepositoryImpl.java

```diff
@@ -0,0 +1,118 @@
+package ca.softwarespace.qiyanna.dataaggregator.repositories.sql;
+
+import static ca.softwarespace.qiyanna.dataaggregator.models.generated.tables.LeagueEntry.LEAGUE_ENTRY;
+
+import ca.softwarespace.qiyanna.dataaggregator.models.dto.LeagueEntryDto;
+import ca.softwarespace.qiyanna.dataaggregator.models.generated.tables.records.LeagueEntryRecord;
+import ca.softwarespace.qiyanna.dataaggregator.repositories.sql.interfaces.SQLLeagueEntryRepository;
+import com.merakianalytics.orianna.types.core.league.LeagueEntry;
+import org.jooq.DSLContext;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+
+@Repository
+public class SQLLeagueEntryRepositoryImpl implements SQLLeagueEntryRepository {
+
+  private final DSLContext dsl;
+
+  @Autowired
+  public SQLLeagueEntryRepositoryImpl(DSLContext dsl) {
+    this.dsl = dsl;
+  }
+
+  @Transactional
+  @Override
+  public LeagueEntryDto add(LeagueEntryDto dto) {
+    LeagueEntryRecord persisted = dsl.insertInto(LEAGUE_ENTRY)
+        .set(createRecord(dto))
+        .returning().fetchOne();
+
+    return convertRecordToDto(persisted);
+  }
+
+  @Transactional
+  @Override
+  public LeagueEntryDto update(LeagueEntryDto dto) {
+    dsl.update(LEAGUE_ENTRY)
+        .set(LEAGUE_ENTRY.QUEUEID, dto.getQueueId())
+        .set(LEAGUE_ENTRY.RANKID, dto.getRankId())
+        .set(LEAGUE_ENTRY.ACCOUNTID, dto.getAccountId())
+        .set(LEAGUE_ENTRY.TIERID, dto.getTierId())
+        .set(LEAGUE_ENTRY.FRESHBLOOD, dto.isFreshBlood())
+        .set(LEAGUE_ENTRY.HOTSTREAK, dto.isHotStreak())
+        .set(LEAGUE_ENTRY.INACTIVE, dto.isInactive())
+        .set(LEAGUE_ENTRY.LEAGUEID, dto.getLeagueId())
+        .set(LEAGUE_ENTRY.LEAGUEPOINTS, dto.getLeaguePoints())
+        .set(LEAGUE_ENTRY.LOSSES, dto.getLosses())
+        .set(LEAGUE_ENTRY.VETERAN, dto.isVeteran())
+        .set(LEAGUE_ENTRY.WINS, dto.getWins()).execute();
+    return findByAccountId(dto.getAccountId());
+  }
+
+  @Transactional(readOnly = true)
+  @Override
+  public LeagueEntryDto findByAccountId(String accountId) {
```

Comment:
This method does more then a repository should do, findByAccountId should just return the element, not do the transformation to the actual DTO.

Throw the result to a service layer and let the service layer use another 'factory'-service to transform the object to the dto.

Reply:
You don't want databases object else where in your code, because that will expose you database. for example, using a record, i can directly affect the database (update an element, delete it, or even creating a new one)  without going through a repository. That may affect the data integrity.

Reply:
Database exposure isnt the issue, based on the result you will know the database setup anyway.

The point is that database objects dont go anywhere else then your repository as the service only transforms them in 1 method, which is the correct way to handle it. Look up the repository pattern people there also return the actual object as it makes room for abstraction https://dev.to/kylegalbraith/getting-familiar-with-the-awesome-repository-pattern--1ao3